src/components/PublicMap/PublishedComponents/DataRecordsList.tsx

```diff
@@ -34,9 +36,25 @@ export default function DataRecordsList({
   const { publicMap, setRecordSidebarVisible } = useContext(PublicMapContext);
   const { mapRef } = useContext(MapContext);
   const { selectedDataRecord } = useContext(DataRecordContext);
+  const { publicFilters, records, setRecords } =
+    useContext(PublicFiltersContext);
   const [expandedRecordId, setExpandedRecordId] = useState<string | null>(null);
 
-  const records = dataRecordsQuery?.data?.dataSource?.records || [];
+  useEffect(() => {
+    const allRecords = dataRecordsQuery?.data?.dataSource?.records || [];
+    const activeFilters = getActiveFilters(publicFilters);
+
+    // if no filters are selected - show all records
+    if (!publicFilters?.length || !activeFilters?.length) {
+      setRecords(allRecords);
+      return;
+    }
+
+    const filteredRecords = filterRecords(activeFilters, allRecords);
+
+    setRecords(filteredRecords);
+  }, [publicFilters, setRecords, dataRecordsQuery?.data?.dataSource?.records]);
```

Comment:
suggestion: this could go in `PublicFiltersProvider` instead of in this component